File: searchInRotatedSortedArray.js
var search = function (nums, target, l, r, mid) {
    nums.sort()
    if (mid < 0 || mid > nums.length-1) return -1;

     mid = Math.floor(nums.length / 2);
     l = nums.slice(0, mid);
     r = nums.slice(mid+1, nums.length)
    
   
        if (nums[mid] === target) {
            return mid;
        } else if (target > nums[mid]) {
             search(nums, target, l, r, mid)
        } else {
            let right = search(nums, target, l, r, mid) 
            if (right === -1) return -1;
            return (right) + mid + 1
        }
    

    
};

console.log(search(nums = [4, 5, 6, 7, 0, 1, 2], target = 6))

File: ruby/search_in_rotated_sorted_array.rb
def search(nums, target)
    return -1 if nums.empty?

    mid = nums.length / 2
    l = nums[0...mid]
    r = nums[mid+1..nums.length]

   

    if (nums[mid] == target)
        return mid
    elsif target < nums[mid]
        return search(l, target)
    else
        right = search(r, target)
        return right if right == -1
        return (right + mid) + 1
    end
end


p search(nums = [4,5,6,7,0,1,2], target = 0)